src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGeneratorTest.java

```diff
@@ -458,4 +461,29 @@ public void testConstructorWithTreeWalkerAuditEvent() {
         assertEquals(expected, actual, "Generated queries do not match expected ones");
     }
 
+    @Test
+    public void testEncode() {
+        final Map<String, String> encoding = new HashMap<>();
+
+        encoding.put("<", "&lt;");
+        encoding.put(">", "&gt;");
+        encoding.put("'", "&apos;&apos;");
+        encoding.put("\"", "&quot;");
+        encoding.put("&", "&amp;");
+        encoding.put("&lt;", "&amp;lt;");
+        encoding.put("abc;", "abc;");
+        encoding.put("&#0;", "&amp;#0;");
+        encoding.put("&#0", "&amp;#0");
+        encoding.put("&#X0;", "&amp;#X0;");
+        encoding.put("\u0001", "#x1;");
+        encoding.put("\u0080", "#x80;");
+        encoding.put("\n", "&#10;");
+        encoding.put("\r", "");
+
+        for (Map.Entry<String, String> entry : encoding.entrySet()) {
+            final String encoded = XpathQueryGenerator.encode(entry.getKey());
```

Comment:
please try to make this method (`encode`) as private and please test it through general execution of checkstyle validation `XpathRegressionMatchXpathTest.java`

Reply:
I believe that this is also resolved since we are no longer implementing check`s similar to regression.

Reply:
please do a test like testTabWidthAfterVoidLiteral
it will a bit verbose but actually same how it will work.
you can put all such small string to single or two String value and validate whole string `"&amp;lt;&amp;#0#x80;"`

Reply:
Done